src/runtime-rs/crates/resource/src/rootfs/nydus_rootfs.rs

```diff
@@ -0,0 +1,169 @@
+                        extra_options.snapshotdir, snapshotdir
+                    )
+                })?;
+                let mut options: Vec<String> = Vec::new();
+                options.push(
+                    "lowerdir=".to_string()
+                        + &do_get_guest_path(
+                            HYBRID_ROOTFS_WRITABLE_LAYER_LOWER_DIR,
+                            cid,
+                            false,
+                            true,
+                        ),
+                );
+                options.push(
+                    "workdir=".to_string()
+                        + &do_get_guest_path(
+                            format!("{}/{}", SNAPSHOTDIR, "work").as_str(),
+                            cid,
+                            false,
+                            false,
+                        ),
+                );
+                options.push(
+                    "upperdir=".to_string()
+                        + &do_get_guest_path(
+                            format!("{}/{}", SNAPSHOTDIR, "fs").as_str(),
+                            cid,
+                            false,
+                            false,
+                        ),
+                );
+                options.push("index=off".to_string());
+                Ok((
+                    Storage {
+                        driver: KATAOVERLAYDEVTYPE.to_string(),
+                        source: TYPE_OVERLAY_FS.to_string(),
+                        fs_type: TYPE_OVERLAY_FS.to_string(),
+                        options,
+                        mount_point: rootfs_guest_path.clone(),
+                        ..Default::default()
+                    },
+                    rootfs_guest_path,
+                ))
+            }
+            // TODO
+            NYDUS_ROOTFS_V6 => {
+                // As some kernel patch is not support in the upstream community, we will delivery this feature later.
+                // see this issue (https://github.com/kata-containers/kata-containers/issues/5143)
+                let errstr: &str = "new_nydus_rootfs: nydus v6 unavailable currently";
+                error!(sl!(), "{}", errstr);
+                Err(anyhow!(errstr))
+            }
+            _ => {
+                let errstr: &str = "new_nydus_rootfs: invalid nydus rootfs type";
+                error!(sl!(), "{}", errstr);
+                Err(anyhow!(errstr))
+            }
+        }?;
+        Ok(NydusRootfs {
+            guest_path: rootfs_guest_path,
+            rootfs: rootfs_storage,
+        })
+    }
+}
+
+#[async_trait]
+impl Rootfs for NydusRootfs {
+    async fn get_guest_rootfs_path(&self) -> Result<String> {
+        Ok(self.guest_path.clone())
+    }
+
+    async fn get_rootfs_mount(&self) -> Result<Vec<oci::Mount>> {
+        todo!()
+    }
+
+    async fn get_storage(&self) -> Option<Storage> {
+        Some(self.rootfs.clone())
+    }
+}
+
+fn get_nydus_extra_options(mount: &Mount) -> Result<NydusExtraOption> {
```

Comment:
Suggest add UT for `get_nydus_extra_options()`.

Reply:
fixed